drafts/25-xxx-specifications.txt

```diff
@@ -413,22 +413,141 @@ in28. Unlike INCLUDE lines (see the section on "INCLUDE lines"),
       same source file at a deeper level of nesting.
 
+
 3.5 The '#if', '#ifdef', '#ifndef', '#elif', '#elifdef',
-      '#elifndef', '#else', '#endif' directives
+      '#elifndef', '#else', '#endif' conditional directives
 ---------------------------------------------------------
 
-Example syntax:
+Example syntax (extra spacing for illustration purposes only):
+    General form:
+        <if-head>                        #endif EOL
+        <if-head>  <if-elif>             #endif EOL
+        <if-head>  <if-elif>  <if-else>  #endif EOL
+
+    Where:
+        <if-head> is one of:
+            #if token-list EOL  <lines>
+            #ifdef ID EOL       <lines>
+            #ifndef ID EOL      <lines>
+
+        <if-elif> is zero or more of:
+            #elif     token-list EOL  <lines>
+            #elifdef  ID EOL          <lines>
+            #elifndef ID EOL          <lines>
 
+        <if-else> is zero or one of:
+            #else EOL  <lines>
+
+        <lines> are zero or more of:
+            directive lines
+            Fortran comment lines
+            Fortran source lines
 
 
 3.5.1 Static semantics specifications
---------------------------------------
+-------------------------------------
+
+if05. A #if, #ifdef, or #ifndef directive signals the start of a list
+      of conditional directives.
+
+if10: Whitespace must immediately follow the directives tokens
+      'ifdef', 'ifndef', 'elifdef', and 'elifndef' to separate them
+      from the ID tokens.
+
+if15. The list of #elif, #elifdef, and #elifndef directives, and the
+      #else directive (if present) are part of the same list of
+      directives introduced by the #if directive.
+
+if20. The chain of conditional directives ends with the #endif
+      directive.
+
+if25. Within a conditional directive list, a #if, #ifdef, or #endif
+      directive introduces a new list of conditional directives, at a new
+      nesting level, within the enclosing conditional directive list.
+
+if30. The conditional directive lists must properly nest.
+
+if35. As shown in the illustrative syntax, there should be no
+      unmatched #endif directives.
+
+if40. As shown in the illustrative syntax, there should be no more
+      than one #else directive in a list of conditional directives at
+      any nesting level.
 
 
 3.5.2 Evaluation semantics specifications
 -----------------------------------------
 
 
+ix05. The #ifdef and #ifndef directives are evaluated as if they had been
+      written "#if defined(ID)" and "#if !defined(ID)" respectively.
+      (For brevity, the descriptions of #ifdef and #ifndef directives
+      below are omitted, and assume this transformation.)
+
+ix10. The #elifdef and #elifndef macros are evaluated as if they had
+      been written "#elif defined(ID)" and "#elif !defined(ID)"
+      respectively.
+
+ix15. The token-lists in #if and #elif are processed as described in
+      the section "Macro recognition and expansion".
```

Comment:
As noted in the meeting, we also need to clarify whether macro expansion occurs within an `#elif` directive line if we already know that "group" is being skipped

Reply:
Added text in today's update. Will request further reviews.